pkg/kubectl/rolling_updater.go

```diff
@@ -171,32 +190,17 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 			oldName, oldRc.Spec.Replicas, 0)
 		oldRc.Spec.Replicas = 0
 		oldRc, err = r.resizeAndWait(oldRc, retry, waitForReplicas)
-		// oldRc, err = r.resizeAndWait(oldRc, interval, timeout)
 		if err != nil {
 			return err
 		}
 	}
-	// add remaining replicas on newRc
-	if newRc.Spec.Replicas != desired {
-		fmt.Fprintf(out, "Resizing %s replicas: %d -> %d\n",
-			newName, newRc.Spec.Replicas, desired)
-		newRc.Spec.Replicas = desired
-		newRc, err = r.resizeAndWait(newRc, retry, waitForReplicas)
-		if err != nil {
-			return err
-		}
-	}
-	// Clean up annotations
-	if newRc, err = r.c.GetReplicationController(r.ns, newName); err != nil {
-		return err
-	}
-	delete(newRc.ObjectMeta.Annotations, sourceIdAnnotation)
-	delete(newRc.ObjectMeta.Annotations, desiredReplicasAnnotation)
-	newRc, err = r.updateAndWait(newRc, interval, timeout)
+	// add remaining replicas on newRc and clean up annotations
+	err = r.resizeAndCleanUp(newRc, newName, desired, retry, waitForReplicas, out)
```

Comment:
I think you need to return/overwrite newRc here. Otherwise, if cleanup policy is to delete/rename, rename will create a copy of the newRc using the outdated spec that still has the old size and replica annotations.